TruncatedChunkException calling HttpResponse.disconnect()

#### Environment details

Using:
com.google.apis:google-api-services-androidpublisher:v3-rev142-1.25.0
com.google.http-client:google-http-client:1.39.2

2. OS type and version: OS Big Sur (11.2.3)
3. Java version: OpenJDK Runtime Environment Corretto-11.0.10.9.1 (build 11.0.10+9-LTS)
4. google-http-client version(s): 1.39.1 and 1.39.2

#### Steps to reproduce

Call com.google.api.client.http.HttpResponse.disconnect() after com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed()

#### Code example

```java
final AndroidPublisher.Inappproducts inAppProductsApi = this.publisherApiClient.inappproducts();
final AndroidPublisher.Inappproducts.List list = inAppProductsApi.list(this.packageName);

final HttpResponse httpResponse = list.executeUnparsed();
final int statusCode = httpResponse.getStatusCode();
httpResponse.disconnect();
```

#### Stack trace
```
http-outgoing-0 >> GET /androidpublisher/v3/applications/{package_name}/inappproducts HTTP/1.1
http-outgoing-0 >> User-Agent: Google-API-Java-Client Google-HTTP-Java-Client/1.39.2 (gzip)
headers|http-outgoing-0 >> Host: www.googleapis.com
headers|http-outgoing-0 >> Connection: Keep-Alive

headers|http-outgoing-0 << HTTP/1.1 200 OK
headers|http-outgoing-0 << Content-Type: application/json; charset=UTF-8
headers|http-outgoing-0 << Content-Encoding: gzip
headers|http-outgoing-0 << Transfer-Encoding: chunked

MainClientExec|Connection can be kept alive indefinitely
MainClientExec|Cancelling request execution
DefaultManagedHttpClientConnection|http-outgoing-0: Shutdown connection
MainClientExec|Connection discarded
PoolingHttpClientConnectionManager|Connection released: [id: 0][route: {s}->https://www.googleapis.com:443][total available: 0; route allocated: 0 of 20; total allocated: 0 of 200]

org.apache.http.TruncatedChunkException: Truncated chunk (expected size: 974; actual size: 595)
	at org.apache.http.impl.io.ChunkedInputStream.read(ChunkedInputStream.java:197) ~[httpcore-4.4.14.jar:4.4.14]
	at org.apache.http.impl.io.ChunkedInputStream.read(ChunkedInputStream.java:210) ~[httpcore-4.4.14.jar:4.4.14]
	at org.apache.http.impl.io.ChunkedInputStream.close(ChunkedInputStream.java:312) ~[httpcore-4.4.14.jar:4.4.14]
	at org.apache.http.impl.execchain.ResponseEntityProxy.streamClosed(ResponseEntityProxy.java:142) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.conn.EofSensorInputStream.checkClose(EofSensorInputStream.java:228) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.conn.EofSensorInputStream.close(EofSensorInputStream.java:172) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.client.entity.LazyDecompressingInputStream.close(LazyDecompressingInputStream.java:97) ~[httpclient-4.5.13.jar:4.5.13]
	at com.google.api.client.http.HttpResponse.ignore(HttpResponse.java:427) ~[google-http-client-1.39.2.jar:1.39.2]
	at com.google.api.client.http.HttpResponse.disconnect(HttpResponse.java:441) ~[google-http-client-1.39.2.jar:1.39.2]
```

#### Any additional information below

Does not happen on 1.39.0. The change that cause the issue was https://github.com/googleapis/google-http-java-client/commit/f84ed5964f376ada5eb724a3d1f3ac526d31d9c5

Alright, anyone can easily reproduce this `https://gcr.io`. (See https://github.com/GoogleContainerTools/jib/issues/3409#issuecomment-898726023 if you think you can't reproduce.)
```java
    new ApacheHttpTransport().createRequestFactory()
        .buildGetRequest(new GenericUrl("https://gcr.io")).execute().disconnect();
```

```
Caused by: org.apache.http.TruncatedChunkException: Truncated chunk (expected size: 2,724; actual size: 229)
    at org.apache.http.impl.io.ChunkedInputStream.read (ChunkedInputStream.java:197)
    at org.apache.http.impl.io.ChunkedInputStream.read (ChunkedInputStream.java:210)
    at org.apache.http.impl.io.ChunkedInputStream.close (ChunkedInputStream.java:312)
    at org.apache.http.impl.execchain.ResponseEntityProxy.streamClosed (ResponseEntityProxy.java:142)
    at org.apache.http.conn.EofSensorInputStream.checkClose (EofSensorInputStream.java:228)
    at org.apache.http.conn.EofSensorInputStream.close (EofSensorInputStream.java:172)
    at org.apache.http.client.entity.LazyDecompressingInputStream.close (LazyDecompressingInputStream.java:97)
    at com.google.api.client.http.HttpResponse.ignore (HttpResponse.java:427)
    at com.google.api.client.http.HttpResponse.disconnect (HttpResponse.java:441)
    at Main.main (Main.java:242)
```

@chingor13 @rawler I think this is a critical bug, and #1303 should be reverted.

--- a/google-http-client-apache-v2/src/test/java/com/google/api/client/http/apache/v2/ApacheHttpTransportTest.java
+++ b/google-http-client-apache-v2/src/test/java/com/google/api/client/http/apache/v2/ApacheHttpTransportTest.java
@@ -321,38 +321,4 @@ public class ApacheHttpTransportTest {
   private boolean isWindows() {
     return System.getProperty("os.name").startsWith("Windows");
   }
-
-  @Test(timeout = 10_000L)
-  public void testDisconnectShouldNotWaitToReadResponse() throws IOException {
-    // This handler waits for 100s before returning writing content. The test should
-    // timeout if disconnect waits for the response before closing the connection.
-    final HttpHandler handler =
-        new HttpHandler() {
-          @Override
-          public void handle(HttpExchange httpExchange) throws IOException {
-            byte[] response = httpExchange.getRequestURI().toString().getBytes();
-            httpExchange.sendResponseHeaders(200, response.length);
-
-            // Sleep for longer than the test timeout
-            try {
-              Thread.sleep(100_000);
-            } catch (InterruptedException e) {
-              throw new IOException("interrupted", e);
-            }
-            try (OutputStream out = httpExchange.getResponseBody()) {
-              out.write(response);
-            }
-          }
-        };
-
-    try (FakeServer server = new FakeServer(handler)) {
-      HttpTransport transport = new ApacheHttpTransport();
-      GenericUrl testUrl = new GenericUrl("http://localhost/foo//bar");
-      testUrl.setPort(server.getPort());
-      com.google.api.client.http.HttpResponse response =
-          transport.createRequestFactory().buildGetRequest(testUrl).execute();
-      // disconnect should not wait to read the entire content
-      response.disconnect();
-    }
-  }
 }

--- a/google-http-client/src/main/java/com/google/api/client/http/HttpResponse.java
+++ b/google-http-client/src/main/java/com/google/api/client/http/HttpResponse.java
@@ -442,10 +442,8 @@ public final class HttpResponse {
    * @since 1.4
    */
   public void disconnect() throws IOException {
-    // Close the connection before trying to close the InputStream content. If you are trying to
-    // disconnect, we shouldn't need to read any further content.
-    response.disconnect();
     ignore();
+    response.disconnect();
   }
 
   /**

